crhchttp/http_cloud_dot_redhat.go

```diff
@@ -113,21 +116,49 @@ func getClient(logger logr.Logger, validateCert bool) http.Client {
 	} else {
 		log.Info("Configured to not upload using the certificate!")
 		// Default the client
-		client := http.Client{}
+		client := http.Client{Timeout: 30 * time.Second}
 		return client
 	}
 }
 
+// ProcessResponse Log response for request and return valid
+func ProcessResponse(logger logr.Logger, resp *http.Response) ([]byte, error) {
+	log := logger.WithValues("costmanagement", "ProcessResponse")
+	// Add error handling and logging here
+	requestID := resp.Header.Get("x-rh-insights-request-id")
+
+	log.Info(fmt.Sprintf("gateway server %s - %s returned %d, x-rh-insights-request-id=%s", resp.Request.Method, resp.Request.URL, resp.StatusCode, requestID))
+	if resp.StatusCode == http.StatusBadRequest || resp.StatusCode >= 300 || resp.StatusCode < 200 {
+		body, _ := ioutil.ReadAll(resp.Body)
+		if len(body) > 1024 {
+			body = body[:1024]
+		}
+		log.Info(fmt.Sprintf("Error Response Body: %s", string(body)))
+		return nil, fmt.Errorf(string(body))
+	}
+
+	if resp.StatusCode >= 200 && resp.StatusCode < 300 {
+		log.Info(fmt.Sprintf("Successfully request x-rh-insights-request-id=%s", requestID))
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			log.Error(err, "Error occurred reading the response body.")
+			return nil, err
+		}
+		return bodyBytes, nil
+	}
+	return nil, fmt.Errorf("Unexpected Response")
+}
+
 // Upload Send data to cloud.redhat.com
-func Upload(logger logr.Logger, costConfig *CostManagementConfig, method string, path string, body *bytes.Buffer, mw *multipart.Writer) (string, metav1.Time, error) {
+func Upload(logger logr.Logger, costConfig *CostManagementConfig, method string, uri string, body *bytes.Buffer, mw *multipart.Writer) (string, metav1.Time, error) {
 	log := logger.WithValues("costmanagement", "Upload")
```

Comment:
Not a blocker and if we wanted to do this it could be in a separate PR - but it might be good to think about the logging in the upload functions that we are using for both ingress and creating sources. For example: 
```
2020-10-20T15:36:31.820-0400    INFO    controllers.CostManagement      OpenShift source type is 1      {"costmanagement": "SourceGetOrCreate"}
2020-10-20T15:36:31.820-0400    INFO    controllers.CostManagement      Configured to not upload using the certificate! {"costmanagement": "GetClient"}
2020-10-20T15:36:31.820-0400    INFO    controllers.CostManagement      Uploading using token authentication    {"costmanagement": "SetupRequest"}
```
The above could be changed to not specifically say upload/Uploading since that could get a little muddled when its being used for creating sources.

Reply:
Pushed a clean up for the logging in these common functions.